flink-runtime/src/main/java/org/apache/flink/runtime/state/internal/InternalAppendingState.java

```diff
@@ -31,4 +31,22 @@
  * @param <SV> The type of elements in the state
  * @param <OUT> The type of the resulting element in the state
  */
-public interface InternalAppendingState<K, N, IN, SV, OUT> extends InternalKvState<K, N, SV>, AppendingState<IN, OUT> {}
+public interface InternalAppendingState<K, N, IN, SV, OUT> extends InternalKvState<K, N, SV>, AppendingState<IN, OUT> {
```

Comment:
It almost seems to me that this methods are not truly specific to merging state and could simply become part of the ``InternalKvState`` interface and the abstract implementations can also move directly to the respective abstract classes ``AbstractRocksDBState`` and ``AbstractHeapState``.

Reply:
I had a second thought about this and I think just adding the methods only in `InternalAppendingState` might be a better choice in the end, because it feels only required here to have a way of manipulating the internal type. Sorry for that :)